src/image.rs

```diff
@@ -1184,6 +1206,87 @@ pub struct ProgressDetail {
     total: Option<u64>,
 }
 
+/// Describes query parameters for DELETE /images/{name}
+/// https://docs.docker.com/engine/api/v1.42/#tag/Image/operation/ImageDelete
+#[derive(Default, Debug)]
+pub struct DeleteOptions {
+
+    params: HashMap<&'static str, String>,
+}
+
+impl DeleteOptions {
+    /// Remove the image even if it is being used by stopped containers or has other tags
+    pub fn force(mut self) -> Self {
+        self.params.insert("force", true.to_string());
+        self
+    }
+
+    /// Do not delete untagged parent images
+    pub fn no_prune(mut self) -> Self {
+        self.params.insert("noprune", true.to_string());
+        self
+    }
+
+    /// serialize options as a string. returns None if no options are defined
+    pub fn serialize(&self) -> Option<String> {
+        if self.params.is_empty() {
+            None
+        } else {
+            Some(
+                form_urlencoded::Serializer::new(String::new())
+                    .extend_pairs(&self.params)
+                    .finish(),
+            )
+        }
+    }
+}
+
+/// Describes query parameters for POST /images/prune
+/// https://docs.docker.com/engine/api/v1.42/#tag/Image/operation/ImagePrune
+#[derive(Default, Debug)]
+pub struct PruneOptions {
+
+    params: HashMap<&'static str, Vec<String>>,
+}
+
+impl PruneOptions {
+
+    /// dangling=<boolean> When set to true (or 1), prune only unused and untagged images.
+    /// When set to false (or 0), all unused images are pruned.
```

Comment:
Why are these method names `set_dangling`, `set_until` and `set_label` instead of just `dangling`, until`, and `label`?

Also, in the Docker API, these are all part of the `filters` parameter. There's a possibility that Docker could add additional top-level parameters to the prune call that are not filters. So I think there's a question of whether we want to expose the fact that these are filters directly to the library client. And then internally, the `params` data member should probably be called `filters` so it's possible to add a `params` data member later, in case other parameters are added by the API.